### PR TITLE
UCT/API: Add API for querying component list and attributes

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -117,6 +117,62 @@ typedef struct uct_md_resource_desc {
 } uct_md_resource_desc_t;
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief UCT component attributres field mask
```
Comment (miked-mellanox) [inline src/uct/api/uct.h]: typo: s/attributres

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9170/ for details (Mellanox internal link).

File: src/uct/api/uct.h [hunk 2]
```diff
@@ -117,6 +117,62 @@ typedef struct uct_md_resource_desc {
...
+     * @ref uct_component_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t               field_mask;
+
+    /** Component name */
+    char                   name[UCT_COMPONENT_NAME_MAX];
+
+    /** Amount of memory-domain resources */
```
Comment (brminich) [inline src/uct/api/uct.h]: minor: probably `number` should be used instead of amount

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9500/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9507/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @tonycurtis @bbenton can u pls review?

**Comment by tonycurtis** [comment]
uct.h:159: which is large enough to hold all memory domain resource entries. After
uct.h:160: the call, this array would be filled with information about existing

would -> will

uct.h:162: In order to allocate such array, it's possible to call

such -> this; it's possible to -> you can 

uct.h:166: array could be allocated with such number of entries, and passed to a

such  -> the returned

uct.h:1100: @return UCS_OK if successful, or nonzero error code in case of failure

nonzero or non-zero?  I see both in the source code (ignoring 3rd party code).

**Comment by yosefe** [comment]
@tonycurtis fixed, changed "non-zero" to "nonzero" in another place as well

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9583/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@tonycurtis @shamisp can you pls take a look?

**Comment by shamisp** [comment]
@tonycurtis is OOO. 👍  from my side.

**Pull request merged**